app/forms/routing/routing_group_duplicator_form.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class Routing::RoutingGroupDuplicatorForm < ApplicationForm
+  attribute :name, :string
+  attribute :id, :integer
+
+  validates :name, :id, presence: true
+
+  validate do
+    errors.add(:name, :invalid) unless RoutingGroup.exists?(id)
+    errors.add(:name, :taken) if RoutingGroup.exists?(name: name)
+  end
+
+  private
+
+  def _save
+    dst = RoutingGroup.create!(
+      name: name
+    )
+    src = RoutingGroup.find(id)
```

Comment:
create method `src_routing_group` that caches instance of `RoutingGroup` similar to `JsonApiModel::OriginatedCps#account`

use it in save method and in validator on line 10